contrib/dracut/90zfs/module-setup.sh.in

```diff
@@ -20,59 +20,30 @@ depends() {
 
 installkernel() {
 	instmods zfs
-	instmods zcommon
-	instmods znvpair
-	instmods zavl
-	instmods zunicode
-	instmods zlua
-	instmods icp
-	instmods spl
-	instmods zlib_deflate
-	instmods zlib_inflate
 }
 
 install() {
-	inst_rules @udevruledir@/90-zfs.rules
-	inst_rules @udevruledir@/69-vdev.rules
-	inst_rules @udevruledir@/60-zvol.rules
-	dracut_install hostid
-	dracut_install grep
-	dracut_install @sbindir@/zgenhostid
-	dracut_install @sbindir@/zfs
-	dracut_install @sbindir@/zpool
-	# Workaround for https://github.com/openzfs/zfs/issues/4749 by
-	# ensuring libgcc_s.so(.1) is included
-	if ldd @sbindir@/zpool | grep -qF 'libgcc_s.so'; then
-		# Dracut will have already tracked and included it
-		:;
-	elif command -v gcc-config >/dev/null 2>&1; then
-		# On systems with gcc-config (Gentoo, Funtoo, etc.):
-		# Use the current profile to resolve the appropriate path
-		s="$(gcc-config -c)"
-		dracut_install "/usr/lib/gcc/${s%-*}/${s##*-}/libgcc_s.so"*
-	elif [ "$(echo /usr/lib/libgcc_s.so*)" != "/usr/lib/libgcc_s.so*" ]; then
-		# Try a simple path first
-		dracut_install /usr/lib/libgcc_s.so*
-	elif [ "$(echo /lib*/libgcc_s.so*)" != "/lib*/libgcc_s.so*" ]; then
-		# SUSE
-		dracut_install /lib*/libgcc_s.so*
-	else
-		# Fallback: Guess the path and include all matches
-		dracut_install /usr/lib*/gcc/**/libgcc_s.so*
-	fi
-	# shellcheck disable=SC2050
-	if [ @LIBFETCH_DYNAMIC@ -gt 0 ]; then
-		for d in $libdirs; do
-			[ -e "$d/@LIBFETCH_SONAME@" ] && dracut_install "$d/@LIBFETCH_SONAME@"
-		done
-	fi
-	dracut_install @mounthelperdir@/mount.zfs
-	dracut_install @udevdir@/vdev_id
-	dracut_install awk
-	dracut_install cut
-	dracut_install tr
-	dracut_install head
-	dracut_install @udevdir@/zvol_id
+	inst_rules \
+		@udevruledir@/90-zfs.rules \
+		@udevruledir@/69-vdev.rules \
+		@udevruledir@/60-zvol.rules
+	
+	dracut_install \
+		@sbindir@/zgenhostid \
+		@sbindir@/zfs \
+		@sbindir@/zpool \
+		@udevdir@/vdev_id \
+		@udevdir@/zvol_id \
+		@mounthelperdir@/mount.zfs \
+		hostid \
+		grep \
+		awk \
+		tr \
+		cut \
+		head
```

Comment:
1. `vdev_id` and `zvol_id` are already picked up by `inst_rules`; there is no need to include them here.
2. `dracut_install` is just a wrapper for `inst_multiple`. Why not use `inst_multiple` directly, which looks more like the other dracut helpers (`inst_libdir_file`, `inst_rules`) anyway?
3. Why are the ZFS utilities given hard-coded paths? It is wrong for two reasons:
    1. On usrmerge systems, it prefers a symlink path rather than a canonical path
    2. The dracut installer already knows how to find executables, let it do its job properly and find the ZFS utilities as well.

Reply:
1. I think we might want to see if there is an alternative to `inst_rules` which can be used because `inst_rules` does not allow for sane failure modes (to cause the initramfs building process to fail in case rules can't be added to it, for instance)
2. I will change this: they're both equivalent so there's no harm
3. While I am merely speculating, I think the dracut installer might not work without hard-coded paths in something like nixos?

Reply:
If the path issue breaks with Nix, then *everything* in dracut is going to break. Deviating from expected practice that hard out to put the onus on Nix, not every upstream vendor of dracut modules.